### PR TITLE
Add Http agent with Combine

### PR DESCRIPTION
**Pull request by dvoineu**
1. All methods from Javascript SDK documentation were transferred for more easier orientation in functionality. Then these methods will be compared with current working version of Javascript SDK and adjusted accordingly.
2. All dependencies from external frameworks (Alamofire and PromiseKit used in old iOS SDK) are removed and replaced by native ios frameworks (URLSession and Combine, accordingly).
3. I started implementing an example App without separation of iOS SDK and Example App. Later, after successful integration and working Example all code will be extracted and separate framework will be created. I propose to create not only POD but also SPM (swift package manager) - native ios package manager, for future distribution.
4. On my next step I will start  implementing helper functions from predicate.js for evaluating predicates.
5. Later I would also recommend realise an Example written in SwiftUI (now we are using UIKit in our Example).

### CODE REVIEW

File: EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+// MARK: - Client
+struct Client: Codable {
+    let browser, device, location, platform: String
+}
+
+// MARK: - Experiment
+struct Experiment: Codable {
+    let web: Web
+    let predicate: ExperimentPredicate
+    let buttonColor, ctaText: ButtonColor?
```
Comment (robertsevernsentient) [inline EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift:44]: These should not be part of the SDK -- they're just example changes I created for the experiment

File: EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    let home, next: ButtonColor?
+
+    enum CodingKeys: String, CodingKey {
+        case web
+        case predicate = "_predicate"
+        case buttonColor = "button_color"
+        case ctaText = "cta_text"
+        case id
+        case paused = "_paused"
+        case home, next
```
Comment (robertsevernsentient) [inline EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift:56]: Likewise, these are example context definitions I made for the experiment -- they will not be in every experiment

File: EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    }
+}
+
+// MARK: - ButtonColor
+class ButtonColor: Codable {
```
Comment (robertsevernsentient) [inline EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift:61]: Nothing in the SDK should be specific to the type of change being made

File: EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+// MARK: - ExperimentPredicate
+struct ExperimentPredicate: Codable {
+    let id: Int?
+    let combinator: String?
+    let rules: [Rule]?
+}
+
+// MARK: - Web
+struct Web: Codable {
```
Comment (robertsevernsentient) [inline EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift:112]: likewise -- this should not be in the SDK

**Review by robertsevernsentient — CHANGES_REQUESTED** [review]

File: EvolvAppExample/EvolvAppExample/EvolvKit/Constants/Config.swift [hunk 5]
```diff
@@ -0,0 +1,26 @@
+//
```
[the remark this answers is not in the document]
Comment (dvoineu) [reply EvolvAppExample/EvolvAppExample/EvolvKit/Constants/Config.swift:1]: A Workspace created and EvolvSDK is extracted as a separate module (framework)

File: EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/EvolvAPI.swift [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+enum EvolvAPI {
+    static let agent = Agent()
+    static let base = URL(string: "https://participants.evolv.ai/v1")!
+}
+
+extension EvolvAPI {
+      
+//    TODO: change Issue to other object from Models
+    static func configuration(eid: String, participantID: String) -> AnyPublisher<[Issue], Error> {
```
[the remark this answers is not in the document]
Comment (dvoineu) [reply EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/EvolvAPI.swift:31]: Implementation changed: parameters removed from function
eid changed to environmentID

File: EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/EvolvHttpClient.swift [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+import Foundation
+import Combine
+
+
+let session = URLSession.shared
+
+// MARK: - Testing URL Responses with different status codes
+//    let url = URL(string: "https://httpbin.org/status/422")!
+
+let url = URL(string: "https://participants.evolv.ai/v1/8b50696b6c/C51EEAFC-724D-47F7-B99A-F3494357F164/configuration.json")!
```
[the remark this answers is not in the document]
Comment (dvoineu) [reply EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/EvolvHttpClient.swift:28]: in the previous version hard-coded URL was added for testing. In the actual version is changed.

File: EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Allocations.swift [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+
+import Foundation
+
+// MARK: - Allocation
+struct Allocation: Codable {
+    let uid, sid, eid, cid: String
```
[the remark this answers is not in the document]
Comment (dvoineu) [reply EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Allocations.swift:24]: for experiment id == eid is used and for environment id == environmentID

File: EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Allocations.swift [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    let home: Home
+    let next: Next
+}
+
+// MARK: - Home
+struct Home: Codable {
+    let ctaText: String
+
+    enum CodingKeys: String, CodingKey {
+        case ctaText = "cta_text"
```
[the remark this answers is not in the document]
Comment (dvoineu) [reply EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Allocations.swift:55]: removed

File: EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift (hunk 1 above)
Comment (dvoineu) [reply EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift:44]: removed

File: EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift (hunk 3 above)
Comment (dvoineu) [reply EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift:61]: also removed

File: EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift (hunk 4 above)
Comment (dvoineu) [reply EvolvAppExample/EvolvAppExample/EvolvKit/Service/HttpClient/Models/Configuration.swift:112]: also removed

**Pull request merged**